source/apiVolontaria/volunteer/permissions.py

```diff
@@ -0,0 +1,18 @@
+from rest_framework import permissions
+
+
+class IsOwnerOrReadOnly(permissions.BasePermission):
+    """
+    Custom permission to only allow owners of an object to edit it.
+
+    This is used for Participation only for now.
+    """
+
+    def has_object_permission(self, request, view, obj):
+        # Read permissions are allowed to any request,
+        # so we'll always allow GET, HEAD or OPTIONS requests.
+        if request.method in permissions.SAFE_METHODS:
+            return True
+
+        # Write permissions are only allowed to the owner of the snippet.
+        return obj.user == request.user
```

Comment:
Can you just explain to me this condition ? I'm really interrested, seems really cool!

Reply:
This is a custom permission. You can see an [example](http://www.django-rest-framework.org/api-guide/permissions/#examples) in the DRF docs.

Here, `obj` is the object we're about to update or delete (a participation in our case). We check if the field `user` of that object is equal to the calling user (`request.user`). If it is the case, the user "owns" the object and consequently has the required "permission" to update or delete the object.

Note that the field `user` must exist in the object and must represent the owner of the object. So this permission only works with `Participation` objects for now.

Reply:
Ok! Thanks for the explanation ;)